tests/unit/frameworks/pytorch/resnet_serialize_test.py

```diff
@@ -46,23 +58,20 @@ def test_identical(self):
             init_batch_norm=False,
             device=device,
         )
-        # create temp file
-        model_path = tempfile.NamedTemporaryFile(delete=True)
-        # and write the created model
-        torch.save(model.state_dict(), model_path)
 
-        model_copy = create_model(
+        checkpoint_file = save_model(model)
+
+        model2 = create_model(
             model_class=model_class,
             model_args=model_args,
             init_batch_norm=False,
             device=device,
+            checkpoint_file=checkpoint_file.name
         )
-        # copy the initial model parameters to the new model
-        model_copy.load_state_dict(torch.load(model_path.name))
 
-        # comapre the initial and new model
-        self.assertTrue(compare_models(model, model_copy, input_shape=(3, 224, 224)))
-        model_path.close()
+        checkpoint_file.close()
+        assert checkpoint_file.file.closed
```

Comment:
Do we need this assert? If so, it should be self.assertTrue()

Reply:
done